src/main/java/com/google/sps/servlets/GenerateServlet.java

```diff
@@ -28,14 +30,162 @@
 
   private static final String FILTER_ARRAY = "clicked-array";
 
+  String[] allPlaces = {"Paris", "New York City", "San Francisco", "London", "Sydney", "Venice"};
+  String[] allDifficulties = {"Easy", "Medium", "Hard"};
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    HashSet<String> userPlaces = new HashSet();
+    HashSet<String> userDifficulties = new HashSet();
+
+    ArrayList<Destination> allDestinations = new ArrayList();
+    // ArrayList<Name of Destination>
+    ArrayList<String> filteredDestinations = new ArrayList();
+
     // Get array of clicked filters and convert to ArrayList<String>
-    Gson gson = new Gson();
     HashSet<String> clickedFilters =
-        gson.fromJson(request.getParameter(FILTER_ARRAY), HashSet.class);
+        new Gson().fromJson(request.getParameter(FILTER_ARRAY), HashSet.class);
+
+    allDestinations = createFakeDestinations();
+    userPlaces = separateFilters(clickedFilters, allPlaces);
+    userDifficulties = separateFilters(clickedFilters, allDifficulties);
+    filteredDestinations = filterPlaces(userPlaces, allDestinations);
+    filteredDestinations =
+        filterDifficulty(filteredDestinations, userDifficulties, allDestinations);
+    writeToDataStore(filteredDestinations);
 
     response.setContentType("text/html;");
     response.getWriter().println(clickedFilters);
   }
+
+  public HashSet<String> separateFilters(HashSet<String> clickedFilters, String[] allTypeFilters) {
+    HashSet<String> userFilters = new HashSet();
+    for (int i = 0; i < allTypeFilters.length; i++) {
+      if (clickedFilters.contains(allTypeFilters[i])) {
+        userFilters.add(allTypeFilters[i]);
+      }
+    }
+    return userFilters;
+  }
+
+  /* Return Destination objects within specified place. */
```

Comment:
Nit: Is returning the names destinations rather than a full destination object, update javadoc.